Add a class_exists check to PDO MySQL driver

mbabker opened the pull request:
There are some reports in the forum about users getting a "class 'PDO' not found" error post upgrade.  This appears to be due to calling a PDO class method which may not be present if PDO isn't supported, so this adds a class_exists check before that call.

Commit:
Add a class_exists check to PDO MySQL driver

stresskills commented:
Thank you! fixed it for me. Any idea why this happened. I don't know much about PDO.

mbabker commented:
> Thank you! fixed it for me. Any idea why this happened. I don't know much about PDO.

We added support for PDO based connections for MySQL.  But, it's an extension that isn't always compiled with PHP.  We simply forgot to add a check for it before trying to run code from the class, and it's something that slipped by while testing (apparently all testing was in environments that have the extension).

ikarkalakos commented:
Worked for me too. Thanks

stresskills commented:
> We added support for PDO based connections for MySQL. But, it's an extension that isn't always compiled with PHP. We simply forgot to add a check for it before trying to run code from the class, and it's something that slipped by while testing (apparently all testing was in environments that have the extension).

Thank you for your response. I will add PDO to my server. I know my friend has been telling me the benefits. I guess time to look into it. Have a great day!

wilsonge commented:
Multiple good tests. Merged

8ofproject commented:
Works like a charm. Thanks.
